pkg/golinters/promlinter.go

```diff
@@ -0,0 +1,74 @@
+package golinters
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+	"sync"
+
+	"github.com/yeya24/promlinter"
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
+	"github.com/golangci/golangci-lint/pkg/lint/linter"
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+func NewPromlinter() *goanalysis.Linter {
+	var mu sync.Mutex
+	var resIssues []goanalysis.Issue
+
+	const linterName = "promlinter"
+	analyzer := &analysis.Analyzer{
+		Name: linterName,
+		Doc:  goanalysis.TheOnlyanalyzerDoc,
+	}
+	return goanalysis.NewLinter(
+		linterName,
+		"Check Prometheus metrics naming via promlint",
+		[]*analysis.Analyzer{analyzer},
+		nil,
+	).WithContextSetter(func(lintCtx *linter.Context) {
+		strict := lintCtx.Cfg.LintersSettings.Promlinter.Strict
+		disabledLinters := lintCtx.Cfg.LintersSettings.Promlinter.DisabledLinters
+
+		analyzer.Run = func(pass *analysis.Pass) (interface{}, error) {
+			files := make([]*ast.File, 0)
+
+			for _, f := range pass.Files {
+				if strings.HasSuffix(pass.Fset.Position(f.Pos()).Filename, "_test.go") {
```

Comment:
Could you explain why you have hardcoded a filter on test files?

Reply:
Oh, I want to skip the test files. Is there any other way to do this?

Reply:
yes there are several other ways:

```yml
# global approach
run:
  tests: false
```

or

```yml
# specific for a linter
issues:
  exclude-rules:
    - path: .*_test.go
      linters:
        - promlinter
```

or add an option to the linter.

Reply:
Thanks, I have updated my code.